pkg/util/client/leaderelectionconfig/config.go

```diff
@@ -69,5 +70,5 @@ func BindFlags(l *options.LeaderElectionConfiguration, fs *pflag.FlagSet) {
 		"of a leadership. This is only applicable if leader election is enabled.")
 	fs.StringVar(&l.ResourceLock, "leader-elect-resource-lock", l.ResourceLock, ""+
 		"The type of resource object that is used for locking during "+
-		"leader election. Supported options are `endpoints` (default) and `configmap`.")
+		"leader election. Supported options are `endpointsleases` (default).")
```

Comment:
We don't have to remove already supported options from the doc string of the flag.

```suggestion
	fs.StringVar(&l.ResourceLock, "leader-elect-resource-lock", l.ResourceLock, ""+
		"The type of resource object that is used for locking during "+
		"leader election. Supported options are 'endpoints', 'configmaps', "+
		"'leases', 'endpointsleases' and 'configmapsleases'.")
```

FYI, this seems to be copied from https://github.com/kubernetes/kubernetes/blob/c75d254beb662ea73736d2f123d0382b7f91dfbd/staging/src/k8s.io/component-base/config/options/leaderelectionconfig.go#L43-L46.